src/drivers/intel/cavs/dmic.c

```diff
@@ -1179,11 +1184,19 @@ static int dmic_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
 		dmic_prm[di]->pdm[i].id = i;
 		for (j = 0; j < config->dmic.num_pdm_active; j++) {
 			/* copy the pdm controller params id the id's match */
-			if (dmic_prm[di]->pdm[i].id == config->dmic.pdm[j].id)
-				memcpy(&dmic_prm[di]->pdm[i],
+			if (dmic_prm[di]->pdm[i].id == config->dmic.pdm[j].id) {
+				ret = memcpy_s(&dmic_prm[di]->pdm[i],
+				       sizeof(
+				       dmic_prm[di]->pdm[i]),
 				       &config->dmic.pdm[j],
 				       sizeof(
 				       struct sof_ipc_dai_dmic_pdm_ctrl));
+				if (ret != 0) {
+					trace_dmic_error("dmic_set_config()"
```

Comment:
small thing but it would be nice to add a space char after f() because this will result in "dmic_set_config()error.." :)